src/model.js

```diff
@@ -14,4 +14,44 @@ function getRandomIndex(array) {
 
 export function getRandomUser(users) {
   return users[getRandomIndex(users)];
+}
+
+export function getAverageWater(hydrationData, id) {
+  const userData = hydrationData.filter(data => data.userID === id); 
+  
+  if (!userData.length) {
+   return 0;
+  }
+
+  return Math.round(userData.reduce((sum, date) => sum + date.numOunces, 0) / userData.length);
+}
+
+export function getDailyWater(hydrationData, id, date) {
+  const userData = hydrationData.find(data => data.userID === id && data.date === date);
+
+  if (!userData) {
+    return 0;
+  }
+
+  return userData.numOunces
+}
+
+export function getWeeklyWater(hydrationData, id) {
+  const userData = hydrationData.filter(data => data.userID === id);
+
+  const week = {};
+
+  if (userData.length === 0) {
+    return week;
+  } else if (userData.length < 7) {
+    for (var i = -1; (userData.length + i) >= 0; i--) {
```

Comment:
Clever use of for loops here! I think this can be done using an assortment of array methods (namely reduce).

Reply:
Smart thinking. I refactored the code and sliced the last 7  logs in the userdata and ran them through reduce in order to clean this code up.